Pidaq/Lib/dts_manager/dts_manager.py

```diff
@@ -0,0 +1,409 @@
+            self.default_can_offset = 160
+            self.temp1_id = self.default_can_offset
+            self.temp2_id = self.default_can_offset + 1
+            self.temp3_id = self.default_can_offset + 2
+            self.analog_inputs_id = self.default_can_offset + 3
+            self.digital_input_status_id = self.default_can_offset + 4
+            self.motor_position_id = self.default_can_offset + 5
+            self.current_info_id = self.default_can_offset + 6
+            self.voltage_info_id = self.default_can_offset + 7
+            self.flux_info_id = self.default_can_offset + 8
+            self.internal_voltages_id = self.default_can_offset + 9
+            self.internal_states_id = self.default_can_offset + 10
+            self.fault_codes_id = self.default_can_offset + 11
+            self.torque_timer_id = self.default_can_offset + 12
+            self.modulation_index_id = self.default_can_offset + 13
+
+        def get_temp1_data(self, start=0, stop=8):
+            return self.bus.messages[self.temp1_id].data[start:stop]
+
+        def get_temp2_data(self, start=0, stop=8):
+            return self.bus.messages[self.temp2_id].data[start:stop]
+
+        def get_temp3_data(self, start=0, stop=8):
+            return self.bus.messages[self.temp3_id].data[start:stop]
+
+        def get_analog_input_voltages_data(self, start=0, stop=8):
+            return self.bus.messages[self.analog_inputs_id].data[start:stop]
+
+        def get_digital_input_status_data(self, start=0, stop=8):
+            return self.bus.messages[self.digital_input_status_id].data[start:stop]
+
+        def get_motor_position_data(self, start=0, stop=8):
+            return self.bus.messages[self.motor_position_id].data[start:stop]
+
+        def get_current_data(self, start=0, stop=8):
+            return self.bus.messages[self.current_info_id].data[start:stop]
+
+        def get_voltage_data(self, start=0, stop=8):
+            return self.bus.messages[self.voltage_info_id].data[start:stop]
+
+        def get_flux_data(self, start=0, stop=8):
+            return self.bus.messages[self.flux_info_id].data[start:stop]
+
+        def get_internal_voltages_data(self, start=0, stop=8):
+            return self.bus.messages[self.internal_voltages_id].data[start:stop]
+
+        def get_internal_states_data(self, start=0, stop=8):
+            return self.bus.messages[self.internal_states_id].data[start:stop]
+
+        def get_faults_data(self, start=0, stop=8):
+            return self.bus.messages[self.fault_codes_id].data[start:stop]
+
+        def get_torque_timer_data(self, start=0, stop=8):
+            return self.bus.messages[self.torque_timer_id].data[start:stop]
+
+        def get_modulation_index_data(self, start=0, stop=8):
+            return self.bus.messages[self.modulation_index_id].data[start:stop]
+
+        def convert_temperatures(self) -> None:
+            if self.get_temp1_data():
```

Comment:
So this checks if there is any data, and the following calls parse specific bytes? Nice

Reply:
Yeah I just have the default params for that to be 0 and 8, so that just checks to see if the whole message contains data, and if the message is `NoneType` then it will just skip over this. Worked out nicely